cmd/otel-allocator/allocation/http.go

```diff
@@ -18,61 +18,24 @@ import (
 	"fmt"
 	"net/url"
 
-	"github.com/prometheus/common/model"
-
 	"github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/target"
 )
 
 type collectorJSON struct {
-	Link string            `json:"_link"`
-	Jobs []targetGroupJSON `json:"targets"`
-}
-
-type targetGroupJSON struct {
-	Targets []string       `json:"targets"`
-	Labels  model.LabelSet `json:"labels"`
+	Link string         `json:"_link"`
+	Jobs []*target.Item `json:"targets"`
 }
 
-func GetAllTargetsByJob(job string, cMap map[string][]target.Item, allocator Allocator) map[string]collectorJSON {
+// GetAllTargetsByJob is a relatively expensive call that is usually only used for debugging purposes.
+func GetAllTargetsByJob(allocator Allocator, job string) map[string]collectorJSON {
 	displayData := make(map[string]collectorJSON)
-	for _, j := range allocator.TargetItems() {
-		if j.JobName == job {
-			var targetList []target.Item
-			targetList = append(targetList, cMap[j.CollectorName+j.JobName]...)
-
-			var targetGroupList []targetGroupJSON
-
-			for _, t := range targetList {
-				targetGroupList = append(targetGroupList, targetGroupJSON{
-					Targets: []string{t.TargetURL},
-					Labels:  t.Label,
-				})
-			}
-
-			displayData[j.CollectorName] = collectorJSON{Link: fmt.Sprintf("/jobs/%s/targets?collector_id=%s", url.QueryEscape(j.JobName), j.CollectorName), Jobs: targetGroupList}
-
-		}
+	for _, col := range allocator.Collectors() {
+		items := allocator.GetTargetsForCollectorAndJob(col.Name, job)
+		displayData[col.Name] = collectorJSON{Link: fmt.Sprintf("/jobs/%s/targets?collector_id=%s", url.QueryEscape(job), col.Name), Jobs: items}
 	}
 	return displayData
 }
 
-func GetAllTargetsByCollectorAndJob(collector string, job string, cMap map[string][]target.Item, allocator Allocator) []targetGroupJSON {
-	var tgs []targetGroupJSON
-	group := make(map[string]target.Item)
-	labelSet := make(map[string]model.LabelSet)
-	if _, ok := allocator.Collectors()[collector]; ok {
-		for _, targetItemArr := range cMap {
-			for _, targetItem := range targetItemArr {
-				if targetItem.CollectorName == collector && targetItem.JobName == job {
-					group[targetItem.Label.String()] = targetItem
-					labelSet[targetItem.Hash()] = targetItem.Label
-				}
-			}
-		}
-	}
-	for _, v := range group {
-		tgs = append(tgs, targetGroupJSON{Targets: []string{v.TargetURL}, Labels: labelSet[v.Hash()]})
-	}
-
-	return tgs
+func GetAllTargetsByCollectorAndJob(allocator Allocator, collector string, job string) []*target.Item {
```

Comment:
in a future PR i'm going to remove this file altogether, for now i'm trying to keep changes to a minimum